src/test/scala/firrtlTests/options/PhaseManagerSpec.scala

```diff
@@ -0,0 +1,223 @@
+// See LICENSE for license details.
+
+package firrtlTests.options
+
+import org.scalatest.{FlatSpec, Matchers}
+
+import firrtl.AnnotationSeq
+import firrtl.options.{Phase, PhaseManager, PhaseManagerException}
+import firrtl.annotations.{Annotation, NoTargetAnnotation}
+
+trait IdentityPhase extends Phase {
+  def transform(annotations: AnnotationSeq): AnnotationSeq = annotations
+}
+
+/** Default [[Phase]] that has no prerequisites and invalidates nothing */
+case object A extends IdentityPhase {
+  override def invalidates(phase: Phase): Boolean = false
+}
+
+/** [[Phase]] that requires [[A]] and invalidates nothing */
+case object B extends IdentityPhase {
+  override def prerequisites: Set[Phase] = Set(A)
+  override def invalidates(phase: Phase): Boolean = false
+}
+
+/** [[Phase]] that requires [[B]] and invalidates nothing */
+case object C extends IdentityPhase {
+  override def prerequisites: Set[Phase] = Set(A)
+  override def invalidates(phase: Phase): Boolean = false
+}
+
+/** [[Phase]] that requires [[A]] and invalidates [[A]] */
+case object D extends IdentityPhase {
+  override def prerequisites: Set[Phase] = Set(A)
+  override def invalidates(phase: Phase): Boolean = phase match {
+    case _: A.type => true
+    case _ => false
+  }
+}
+
+/** [[Phase]] that requires [[B]] and invalidates nothing */
+case object E extends IdentityPhase {
+  override def prerequisites: Set[Phase] = Set(B)
+  override def invalidates(phase: Phase): Boolean = false
+}
+
+/** [[Phase]] that requires [[B]] and [[C]] and invalidates [[E]] */
+case object F extends IdentityPhase {
+  override def prerequisites: Set[Phase] = Set(B, C)
+  override def invalidates(phase: Phase): Boolean = phase match {
+    case _: E.type => true
+    case _ => false
+  }
+}
+
+
+/** [[Phase]] that requires [[C]] and invalidates [[F]] */
+case object G extends IdentityPhase {
+  override def prerequisites: Set[Phase] = Set(C)
+  override def invalidates(phase: Phase): Boolean = phase match {
+    case _: F.type => true
+    case _ => false
+  }
+}
+
+case object CyclicA extends IdentityPhase {
+  override def prerequisites: Set[Phase] = Set(CyclicB)
+}
+
+case object CyclicB extends IdentityPhase {
+  override def prerequisites: Set[Phase] = Set(CyclicA)
+}
+
+class CyclicInvalidateFixture {
+
+  case object A extends IdentityPhase {
+    override def invalidates(phase: Phase): Boolean = false
+  }
+  case object B extends IdentityPhase {
```

Comment:
Its tough to remember what all of these do. Can you give a clearer name, like A=A, B=RecA, C=RecAInvB, etc.?